heareval/tasks/util/dataset_builder.py

```diff
@@ -0,0 +1,209 @@
+"""
+A builder class that helps to construct luigi dataset pre-processing pipelines
+"""
+import os
+from urllib.parse import urlparse
+from typing import Any, Dict, List, Union
+from types import new_class
+from functools import partial
+
+import luigi
+
+from heareval.tasks.config import DatasetConfig, PartitionedDatasetConfig, get_config
+import heareval.tasks.util.luigi as luigi_util
+
+
+class DatasetBuilder:
+    """
+    A class that helps to construct a data preprocessing pipeline for a
+    dataset.
+
+    Args:
+        task_config: the name of the task being built (will try to load the dataset
+            config file for this automatically, or a DatasetConfig file describing the
+            dataset that is being constructed.
+    """
+
+    def __init__(self, task_config: Union[str, DatasetConfig]):
```

Comment:
maybe just get rid of this option to pass in a str and just pass in the config object